app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,74 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, only: %i[new create]
```

Comment:
You are required to create your own before_action helper methods instead of using the one provided by devise gem. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from Odin.